codegen/CodeGenerator/src/main/java/RelationProcessFunctionWriter.java

```diff
@@ -0,0 +1,98 @@
+import org.ainslec.picocog.PicoWriter;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+public class RelationProcessFunctionWriter implements CodeGenerator {
+    private final String className;
+    private final String relationName;
+    private final PicoWriter writer;
+    private final RelationProcessFunction relationProcessFunction;
+
+    public RelationProcessFunctionWriter(final RelationProcessFunction relationProcessFunction) {
+        requireNonNull(relationProcessFunction);
+        this.relationProcessFunction = relationProcessFunction;
+        this.relationName = relationProcessFunction.getName();
+        this.className = relationProcessFunction.getName() + "ProcessFunction";
+        this.writer = new PicoWriter();
+    }
+
+    @Override
+    public void generateCode(final String filePath) throws IOException {
+        CheckerUtils.checkNullOrEmpty(filePath, "filePath");
+        addImports();
+        makeConstructor();
+        makeIsValidFunction(relationProcessFunction.getSelectConditions(), "value");
+        writer.writeln_r("}");
+        Files.write(Paths.get(filePath + File.separator + className + ".scala"), writer.toString().getBytes());
+    }
+
+    private void makeConstructor() {
+
+        String stringBuilder = "class " + className + " extends RelationFKProcessFunction[Any](\"" +
+                relationName + "\"," +
+                "Array()" + "," +
+                "Array()" + "," +
+                "true)" +
+                "{";
+        writer.writeln(stringBuilder);
+    }
+
+    private void makeIsValidFunction(List<SelectCondition> selectConditions, String param) {
+        writer.writeln_r("override def isValid(value: Payload): Boolean = {");
+        StringBuilder ifCondition = new StringBuilder();
+        ifCondition.append("if(");
+        SelectCondition condition;
+        for (int i = 0; i < selectConditions.size(); i++) {
+            ifCondition.append(param);
+            condition = selectConditions.get(i);
+            //Ensure that in the values list the attribute value is always added first
+            Expression expression = condition.getExpression();
```

Comment:
This path is weird. Can you just do a type check and decide how to output?  Since we will have attribute/attribute constant/constant as select condition.  Moreover, you should make the generator for selection as a function (maybe a member function for class select condition or class Expression?)